pkgs/development/compilers/zig/0.13/default.nix

```diff
@@ -0,0 +1,114 @@
+{
+  lib,
+  callPackage,
+  cmake,
+  coreutils,
+  fetchFromGitHub,
+  libxml2,
+  llvmPackages,
+  stdenv,
+  testers,
+  zlib,
+  util-linux
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "zig";
+  version = "0.13.0";
+
+  src = fetchFromGitHub {
+    owner = "ziglang";
+    repo = "zig";
+    rev = finalAttrs.version;
+    hash = "sha256-5qSiTq+UWGOwjDVZMIrAt2cDKHkyNPBSAEjpRQUByFM=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    (lib.getDev llvmPackages.llvm)
+    util-linux
+  ];
+
+  buildInputs =
+    [
+      libxml2
+      zlib
+    ]
+    ++ (with llvmPackages; [
+      libclang
+      lld
+      llvm
+    ]);
+
+  outputs = [
+    "out"
+    "doc"
+  ];
+
+  cmakeFlags = [
+    # file RPATH_CHANGE could not write new RPATH
+    (lib.cmakeBool "CMAKE_SKIP_BUILD_RPATH" true)
+    # ensure determinism in the compiler build
+    (lib.cmakeFeature "ZIG_TARGET_MCPU" "baseline")
+    # always link against static build of LLVM
+    (lib.cmakeBool "ZIG_STATIC_LLVM" true)
+  ];
+
+  env.ZIG_GLOBAL_CACHE_DIR = "$TMPDIR/zig-cache";
+
+  doInstallCheck = true;
+
+  # strictDeps breaks zig when clang is being used.
+  # https://github.com/NixOS/nixpkgs/issues/317055#issuecomment-2148438395
+  strictDeps = !stdenv.cc.isClang;
+
+  # Zig's build looks at /usr/bin/env to find dynamic linking info. This doesn't
+  # work in Nix's sandbox. Use env from our coreutils instead.
+  postPatch = ''
+    substituteInPlace lib/std/zig/system.zig \
+      --replace "/usr/bin/env" "${lib.getExe' coreutils "env"}"
+  '';
+
+  postBuild = ''
+    mkdir -p codedir cache-root/tmp
+    ln -s ../../../doc cache-root/tmp/doc
+
+    for file in ../doc/langref/*.zig; do
+      stage3/bin/zig run ../tools/doctest.zig -- --zig $PWD/stage3/bin/zig --cache-root cache-root -i "$file" -o "codedir/$(basename "$file" | rev | sed 's/giz./tuo./' | rev)"
+    done
+
+    stage3/bin/zig run ../tools/docgen.zig -- --code-dir codedir ../doc/langref.html.in langref.html
```

Comment:
why have you done this? `zig build` already builds the langref.

it's not wise to duplicate the upstream's build logic in shell scripts in the nix package.

Reply:
Because CMake doesn't do it.

Reply:
`stage3/bin/zig build langref`

Reply:
Alright, didn't know that was a step in Zig's `build.zig`.